tests/identity_tests.rs

```diff
@@ -0,0 +1,465 @@
+        "9D61B19DEFFD5A60BA844AF492EC2CC44449C5697B326919703BAC031CAE7F60"
+    ));
+
+    let mut hasher = Sha256::new();
+    hasher.update(b"Moscow");
+
+    // read hash digest and consume hasher
+    let result = hasher.finalize().as_slice().try_into().expect("Wrong size");
+    let claim_data = ClaimData {
+        hashed_info: BTreeSet::from([result]),
+        issuance_date: DATE,
+        valid: true,
+    };
+
+    let claim = Claim {
+        issuer: subject_pair.public().0,
+        issuer_signature: subject_pair.sign(claim_data.encode().as_slice()).0,
+        subject: subject_pair.public().0,
+        verifiers: BTreeMap::new(),
+        data: claim_data.clone(),
+    };
+
+    issue_claim_utils(&id_program, USER, claim.clone(), PIECE_ID, false);
+    let mut claims: BTreeMap<PieceId, Claim> = BTreeMap::new();
+    claims.insert(PIECE_ID, claim.clone());
+    check_user_claims_state_utils(&id_program, subject_pair.public().0, claims);
+    check_claim_state_utils(&id_program, subject_pair.public().0, PIECE_ID, claim);
+    check_date_state_utils(&id_program, subject_pair.public().0, PIECE_ID, DATE);
+
+    let verifier_pair = Sr25519Pair::from_seed(&hex!(
+        "9D61B19DEFFD5D60BA844AF492EC2CC44449C5697B326919703BAC031CAE7F60"
+    ));
+    // verify user with no claims
+    verify_claim_utils(
+        &id_program,
+        USER,
+        verifier_pair.public().0,
+        verifier_pair.sign(claim_data.encode().as_slice()).0,
+        verifier_pair.public().0,
+        PIECE_ID,
+        true,
+    );
+    // verify wrong piece_id
+
+    verify_claim_utils(
+        &id_program,
+        USER,
+        verifier_pair.public().0,
+        verifier_pair.sign(claim_data.encode().as_slice()).0,
+        subject_pair.public().0,
+        PIECE_ID + 1,
+        true,
+    );
+}
+
+#[test]
+fn check_claim() {
+    let sys = System::new();
+    let id_program = init_identity(&sys, USER);
+    let subject_pair = Sr25519Pair::from_seed(&hex!(
+        "9D61B19DEFFD5A60BA844AF492EC2CC44449C5697B326919703BAC031CAE7F60"
+    ));
+
+    let mut hasher = Sha256::new();
+    hasher.update(b"Moscow");
+
+    // read hash digest and consume hasher
+    let city = hasher.finalize().as_slice().try_into().expect("Wrong size");
+    let mut hasher = Sha256::new();
+    hasher.update(b"Nikolskaya");
+    let street = hasher.finalize().as_slice().try_into().expect("Wrong size");
+
+    let claim_data = ClaimData {
+        hashed_info: BTreeSet::from([city, street]),
+        issuance_date: DATE,
+        valid: true,
+    };
+
+    let claim = Claim {
+        issuer: subject_pair.public().0,
+        issuer_signature: subject_pair.sign(claim_data.encode().as_slice()).0,
+        subject: subject_pair.public().0,
+        verifiers: BTreeMap::new(),
+        data: claim_data,
+    };
+
+    issue_claim_utils(&id_program, USER, claim.clone(), PIECE_ID, false);
+    let mut claims: BTreeMap<PieceId, Claim> = BTreeMap::new();
```

Comment:
No need to set the type explicitly.
```suggestion
    let mut claims = BTreeMap::new();
```